app/helpers/random_assign_helper.rb

```diff
@@ -92,38 +96,32 @@ def generate_group_to_students_map(group_id_list)
     end
   end
 
-  def perform_peer_review_assignments
+  # Updates the data structures so that @assigned_reviewer_to_shuffled_reviewee
+  # contains the correct mapping of reviewer to reviewee groups allows creation
+  # of peer reviews, or throws an UnableToRandomlyAssignGroupException if it is
+  # not possible.
+  def perform_peer_review_assignment_mapping
     shuffle_index = 0
+
     while @eligible_reviewer_groups.any?
       reviewer_id = @eligible_reviewer_groups.pop
 
       unless attempt_assign_at_index(reviewer_id, shuffle_index) ||
-             attempt_assign_forward(reviewer_id, shuffle_index) ||
-             attempt_assign_backwards(reviewer_id, shuffle_index)
+             attempt_assign_forward_swap(reviewer_id, shuffle_index) ||
+             attempt_assign_backward_swap(reviewer_id, shuffle_index)
         raise UnableToRandomlyAssignGroupException
       end
 
       shuffle_index += 1
     end
-
-    create_peer_reviews_in_database
   end
 
   def can_assign?(reviewer_id, reviewee_id)
-    if is_reviewer_assigned_to?(reviewer_id, reviewee_id)
-      return false
-    end
-
-    if groups_share_students?(reviewer_id, reviewee_id)
-      # If they do overlap with students, then we know they are never eligible
-      # again, so we should then remember these cannot be assigned to each other.
-      @reviewers_already_assigned_to[reviewer_id].add(reviewee_id)
-      return false
-    end
-
-    true
+    !(is_reviewer_assigned_to?(reviewer_id, reviewee_id) || groups_share_students?(reviewer_id, reviewee_id))
```

Comment:
I decided to change the above since after going over it, I realized that it's technically misleading to store it in `@reviewers_already_assigned_to` when it's not actually assigned to. I also figure that the amount of times `group_share_students(...)` is called should not be that taxing (or if it is, then a change can be made to either revert this, or put it in a set of groups that do not match and update the logic accordingly which probably would be the more clearer option).
If you want, I could add in the last option.